macros/generate_audit_report.sql

```diff
@@ -0,0 +1,95 @@
+{% macro generate_audit_report(model_list, compare_database, compare_schema, identifiers = {}, primary_keys = {}, exclude_columns = {}, audit_helper_macro_name = 'compare_relations') %}
+    
+    {% set compatible_audit_helper_macros = ['compare_relations', 'compare_all_columns', 'compare_relation_columns'] %}
+    {% if audit_helper_macro_name not in compatible_audit_helper_macros %}
+        {{ exceptions.raise_compiler_error("Invalid audit_helper_macro input. Acceptable inputs: " ~ compatible_audit_helper_macros) }}
+    {% endif %}
+
+    {% if audit_helper_macro_name == 'compare_all_columns' and not primary_keys|length %}
+        {{ exceptions.raise_compiler_error("Audit helper macro compare_all_columns requires primary_keys input.") }}
+    {% endif %}
+
+    {% if audit_helper_macro_name == 'compare_relation_columns' %}
+        {% if exclude_columns|length %}
+            {{ exceptions.raise_compiler_error("Audit helper macro compare_relation_columns does not accept exclude_columns input.") }}
+        {% elif primary_keys|length %}
+            {{ exceptions.raise_compiler_error("Audit helper macro compare_relation_columns does not accept primary_keys input.") }}
+        {% endif %}
+    {% endif %}
+
+    with
+
+    {% if execute %}
+
+    {% for m in model_list %}
+
+        {% set m_id = 'model.' ~ project_name ~ '.' ~ m %}
+        {% set m_details = graph["nodes"][m_id] %}
+
+    {{ m }}_results as (
+        {#- *---------------------------------------------
+            * This is where the audit_helper macro is used!
+        --------------------------------------------------* -#}
+
+        {% if identifiers[m] %}
+            {% set legacy_model_name = identifiers[m] %}
+        {% else %}
+            {% set legacy_model_name = m %}
+        {% endif %}
+
+        {% set compare_relation = adapter.get_relation(
+            database = compare_database,
+            schema = compare_schema,
+            identifier = legacy_model_name
+        ) -%}
+
+        {% set dbt_relation = adapter.get_relation(
```

Comment:
@b-per Curious if you think if would be better to use `ref()` instead of `get_relation()` for the dbt_relation? This would require that you add the dummy `-- depends_on: {{ ref('model_a') }}` to your audit report models. I initially didn't enforce this dependency, because it would be annoying to have to list out the `depends_on` for every single model in your `model_list` and I thought it was likely that you would use this macro to generate one-off reports instead of actually including it within your dbt project. The other thing I could try would be dynamically adding the dummy comments to the top of the file via the macro:
```
{% for m in model_list %}
-- depends_on: {{ ref('m') }}`
...
```
but I'm not 100% sure that would work as expected.

Reply:
What would be the advantage of using `ref()` instead of `adapter.get_relation`? So that it could be a model built only after the the upstream models?

Thinking about it, and I think, related to your question, we should maybe clarify that the inputs to the new macros are strings rather than relations (I believe that all the other macros of this package accept relations as input, not just strings describing those relation).

The only difference that comes to mind to me then between `ref()` and `adapter.get_relation()` would be when a model has an alias defined, different from the model name. And this example makes me think that for those we would like to use `ref()` rather than `adapter.get_relation()`. Looping to add `-- depends_on:` shouldn't be a problem.

Reply:
But then, for `generate_audit_report_for_folder`, you can't use relations/dependencies and need to use `adapter.get_relation()` as the list of models is only known after looking at the `graph` object, not available at parse time.

Reply:
> What would be the advantage of using ref() instead of adapter.get_relation? So that it could be a model built only after the the upstream models?
Yes, just to create dependencies. Since it's not possible for `generate_audit_report_for_folder`, I'm thinking that I will just clarify that the input to the new macros are strings rather than relations (so if they want to enforce dependencies they will have to add dummy `-- depends_on:` comments. For alias, I should just be able to check if the model has an alias (from the graph object) and if so use that instead of the name for `adapter.get_relation()`. The other option would be having different behavior for `generate_audit_report_for_folder` vs. `generate_audit_report`. What do you think?

Reply:
I think it makes more sense to have the 2 functions accepting the same inputs (in that case, strings) and just call out in the docs that is is different from the other macros in the package

Reply:
I updated the docs and left it as is! Let me know what you think